File: 프로그래머스/1/176963. 추억 점수/추억 점수.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<string> name, vector<int> yearning, vector<vector<string>> photo) {
    vector<int> res;
    for(int i = 0; i < photo.size(); i++){
        int sum = 0;
        for(int j = 0; j < photo[i].size();j++){
            int f = -1;
            for(int k = 0; k < name.size(); k++){
                if(name[k] == photo[i][j]) {
                    f = k;
                    break;
                }
            }
            if(f != -1) {
               sum += yearning[f]; 
            }
        }
        res.push_back(sum);
    }
    return res;
}

File: 프로그래머스/1/176963. 추억 점수/추억 점수.js
function solution(name, yearning, photo) {
    const y_map = name.reduce((acc,cur,idx)=>{
        return {...acc, [cur]:yearning[idx]}
    },{})
    
    return photo.map(v => v.reduce((acc,cur)=> {
       return acc + (y_map[cur] || 0);
    },0))
}